server/src/main/java/org/opensearch/common/util/ByteUtils.java

```diff
@@ -61,6 +61,16 @@ public static void writeLongLE(long l, byte[] arr, int offset) {
         assert l == 0;
     }
 
+    /** Convert long to a byte array in big-endian format */
+    public static byte[] toByteArrayBE(long l) {
```

Comment:
Lets add unit tests for this method.

Reply:
In the current version of this PR, this is not needed, but it will be for the data integrity delegation change we are making as this will be need for passing the locally calculated checksum to S3.